data_structures/src/Stack/Stack.cs

```diff
@@ -6,44 +6,71 @@ namespace ADT
 {
     public class Stack<T> : IStack<T>
     {
+        private T[] _stack { get; set; }
+        private int _lastIndex { get; set; }
+
         public Stack(int size)
         {
-
+            _stack = new T[size]; 
         }
 
         public IEnumerator<T> GetEnumerator()
         {
-            throw new NotImplementedException();
+             return (IEnumerator<T>)_stack.GetEnumerator(); 
         }
 
         public bool IsEmpty()
         {
-            throw new NotImplementedException();
+            return this.Size() > 0 ? false : true;
         }
 
         public T Peek()
         {
-            throw new NotImplementedException();
+            if(!this.IsEmpty())
+            {
+                return _stack[_lastIndex -1];
+            }
+            else
+            {
+                throw new Exception("Stack is empty- Cannot peek.");
```

Comment:
Nitpick: change `"Stack is empty- Cannot peek."` to `"Stack is empty - Cannot peek."`